src/utils/event.ts

```diff
@@ -42,3 +47,30 @@ export const eventHappeningNow = (
       event.start_utc_seconds + event.duration_minutes > currentTimeInUTCSeconds
   );
 };
+
+export const eventStartTime = (event: VenueEvent) =>
+  fromUnixTime(event.start_utc_seconds);
+
+export const eventEndTime = (event: VenueEvent) =>
+  addMinutes(eventStartTime(event), event.duration_minutes);
+
+export const isEventStartingSoon = (event: VenueEvent) =>
+  differenceInHours(eventStartTime(event), Date.now()) > 0;
+
+export const getEventInterval = (event: VenueEvent) => ({
+  start: eventStartTime(event),
+  end: eventEndTime(event),
+});
+
+export const isEventWithinDate = (checkDate: number | Date) => (
+  event: VenueEvent
+) => {
+  const checkDateInterval = {
+    start: startOfDay(checkDate),
+    end: endOfDay(checkDate),
+  };
```

Comment:
[NB] This could be another useful helper function to extract: `getDayInterval(date)` (in `utils/time`), then use it here.

Reply:
created: https://github.com/sparkletown/sparkle/pull/1426/commits/64e2b9d2c1f4b2cf715ff86330b85947d012a966